### ISSUE TITLE
Logging Property cloud_roleName gets and retains "-Staging" postfix indefinitely after swap

### ISSUE THREAD

**Issue by doreenali** [Azure/azure-webjobs-sdk#1366]
I have a deployment with 8 or so function apps running in production slots. I deployed new code (via an VSTS Build template) to the Staging slot of each function app. Within an hour or so, I swapped the slots and deleted the Staging slot.

In my ApplicationInsights logs, before my deployment, my functions app(s) had a cloud_roleName of "myfunctionapp." After deployment, the cloud_roleName for functions now running in my production slot were "myfunctionapp-staging." Some of my function apps don't run often, and those eventually went back to just "myfunctionapp." However, one of my functions, a trigger that runs on the minute, continued to have the "-staging" postfix until about 8 days later when it went back to "myfunctionapp".

My assumption is that when my other function app instances started back up after inactivity, they resolved to the correct name. But my one function instance that was active enough to not restart, was unable to resolve the name...until one day it was...

My expectations are that something running in the Staging slot would have the "-staging" postfix and something running in the production slot would not.

**Comment by paulbatum** [comment]
@brettsam Is it possible that App Insights is using WEBSITE_HOSTNAME to populate the cloud_roleName property and as we just discovered this can be wrong in the case of slots? (see [here](https://github.com/Azure/azure-webjobs-sdk/issues/1369))

**Comment by brettsam** [comment]
Yes, that's likely it. We're populating this via the `AzureWebAppRoleEnvironmentTelemetryInitializer` that App Insights provides, and it's using WEBSITE_HOSTNAME. We should be able to work around this, though.

https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/WindowsServer/WindowsServer.Shared/AzureWebAppRoleEnvironmentTelemetryInitializer.cs#L17

**Comment by HallJeremyG** [comment]
Any update on this?  Restarting a production web app to fix the name in app insights is not acceptable.

**Comment by brettsam** [comment]
I don't think we realized this was still an issue -- we no longer rely on WEBSITE_HOSTNAME (see here -- https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsRoleEnvironmentTelmetryInitializer.cs).

After the slot swap, are you able to see the values of WEBSITE_SITE_NAME and WEBSITE_SLOT_NAME? They should be updated to the new values and we should be updating all telemetry that flows through.

**Comment by mathewc** [comment]
FYI - we've made recent changes to address the original issue. I made Functions Runtime changes https://github.com/Azure/azure-functions-host/pull/4338 to update the hostname based on incoming requests. ANT 83 will also include updates to ping the site root after a swap, which coupled with the above fix will mean the runtime has the updated value. To get the updated value, code should use `HostNameProvider`.
